qrand/platforms/qsharp/circuit.py

```diff
@@ -28,76 +28,129 @@
 ###############################################################################
 class QsharpCircuit(QuantumCircuit):
     def __init__(self, num_qubits: int) -> None:
-        self.ERROR_MSG = f"{self.__class__.__name__}"  # TODO
-        raise NotImplementedError(self.ERROR_MSG)
+        self.gates: str = ""
+        self._num_qubits: int = num_qubits
+
+    def _validate_qubit_index(self, qubit_index: int) -> None:
+        if qubit_index >= self.num_qubits:
+            raise ValueError(
+                f"Qubit index out of range {qubit_index} >= {self.num_qubits}."
+            )
 
     @property
     def num_qubits(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+        return self._num_qubits
 
     ############################## SPECIAL GATES ##############################
     def measure(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        self._validate_qubit_index(target_qubit)
+        self.gates += f"""
+        if M(q[{target_qubit}]) == One{{
+        set value w/= {target_qubit} <- "1";
+        }}
+        """
 
     ########################### SINGLE QUBIT GATES ###########################
     def h(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        self._validate_qubit_index(target_qubit)
+        self.gates += f"H(q[{target_qubit}]);"
 
     def rx(self, radians: float, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        self._validate_qubit_index(target_qubit)
+        self.gates += f"Rx({radians},q[{target_qubit}]);"
```

Comment:
Does the `radians` value get validated anywhere?

Reply:
No, not yet.

Reply:
Do we need to make sure that `-pi <= radians < pi` or simply validate type `float` @crazy4pi314?